primitives/core/src/bls.rs

```diff
@@ -0,0 +1,780 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Simple BLS (Boneh–Lynn–Shacham) Signature API.
+
+#[cfg(feature = "std")]
+use crate::crypto::Ss58Codec;
+#[cfg(feature = "full_crypto")]
+use crate::crypto::{DeriveError, DeriveJunction, Pair as TraitPair, SecretStringError};
+use crate::{
+	crypto::{ByteArray, CryptoType, Derive, Public as TraitPublic, UncheckedFrom},
+	hash::{H1152, H896},
+};
+
+#[cfg(feature = "full_crypto")]
+use sp_std::vec::Vec;
+
+#[cfg(feature = "full_crypto")]
+use bls_like::{
+	double::{DoublePublicKey, DoublePublicKeyScheme, DoubleSignature},
```

Comment:
"double" is a confusing name for what this is doing.

Reply:
exposed directly from root removed `double`.